### DISCUSSION TITLE
I'm getting "The function identifier is empty" exception while importing plugin from directory.

### DISCUSSION THREAD

**Discussion by abhinav-jha-dev** [microsoft/semantic-kernel#7296] [Q&A] [answered]
I'm trying to import plugin from prompts directory.
<img width="947" alt="image" src="https://github.com/user-attachments/assets/e4207015-0a2c-49ee-8819-374017849d5c">
I've created config.json as:
{
    "schema": 1,
    "type": "completion",
    "description": "Get user intent.",
    "execution_settings": {
        "default": {
            "max_tokens": 1000,
            "temperature": 0
        }
    }
}
And the skprompt.txt
<message role="system">Instructions: What is the intent of this request?
  Do not explain the reasoning, just reply back with the intent. If you are unsure, reply with {{choices.[0]}}.
  Choices: {{choices}}.</message>

  <message role="user">{{request}}</message>
  <message role="system">Intent:</message>

Need help to figure out the issue.

**Comment by dmytrostruk** [comment, marked as answer]
Hi @abhinav-jha-dev ! It looks like the syntax in `skprompt.txt` file is a little bit incorrect. 
The variables should start with `$` sign, so in your case: `{{$choices}}`, `{{$request}}`. Also, I'm not sure there is a support for indexes, so instead of accessing first choice with `{{choices.[0]}}`, you will need to define separate variable in `KernelArguments`, for example `firstChoice`, assign it from code and then access it from prompt with `{{$firstChoice}}`.

**Comment by abhinav-jha-dev** [comment]
Thanks for identifying this. I was able to resolve it using $ for variable data.
